pkgs/desktops/gnome/extensions/pano/default.nix

```diff
@@ -0,0 +1,110 @@
+{ lib
+, stdenv
+, atk
+, cogl
+, fetchFromGitHub
+, glib
+, gnome
+, gobject-introspection
+, gsound
+, gtk3
+, libgda
+, mkYarnModules
+, nodePackages
+, pango
+, substituteAll
+, yarn
+}:
+
+let
+  version = "10";
+  uuid = "pano@elhan.io";
+  pname = "gnome-shell-extension-pano";
+
+  girpathsPatch = (substituteAll {
+    src = ./set-dirpaths.patch;
+    gda_path = "${libgda}/lib/girepository-1.0";
+    gsound_path = "${gsound}/lib/girepository-1.0";
+  });
+in
+stdenv.mkDerivation rec {
+  inherit pname version;
+  src = fetchFromGitHub {
+    owner = "oae";
+    repo = "gnome-shell-pano";
+    rev = "v${version}";
+    hash = "sha256-1vGiWSXlQ8xheqJsZm3uXCixuLl5NFM2OgPQrzCPhME=";
+  };
+
+  nativeBuildInputs = [
+    gobject-introspection
+    nodePackages.rollup
+    nodePackages.yarn
+  ];
+
+  buildInputs = [
+    atk
+    cogl
+    glib
+    gnome.gnome-shell
+    gnome.mutter
+    gsound
+    gtk3
+    libgda
+    pango
+  ];
+
+  nodeModules =
+    mkYarnModules {
+      inherit pname version;
+      name = "${pname}-modules-${version}";
+      packageJSON = "${src}/package.json";
+      yarnLock = "${src}/yarn.lock";
+      yarnNix = ./yarn.nix;
```

Comment:
It is not clear how to update this. Maybe add an `passthru.updateScript`. Did not test it but something like the following should work:

```nix
  passthru = {
    lockFileNix = runCommand
      "pano-yarn.nix"
      {
        nativeBuildInputs = [
          yarn2nix-moretea
        ];
      }
      ''
        unpackPhase
        cd "''${sourceRoot:-.}"
        yarn2nix > "$out"
      '';
    updateScript =
      let
        updateSource = gitUpdater { };
        updateLockFile = _experimental-update-script-combinators.copyAttrOutputToFile "gnomeExtensions.pano.lockFileNix" ./yarn.nix;
      in
      _experimental-update-script-combinators.sequence [
        updateSource
        updateLockFile
      ];
  };
```